.github/workflows/pr.yaml

```diff
@@ -0,0 +1,24 @@
+# This is a template `pr.yaml` file for ops charms
+# This file is managed by bootstack-charms-spec and should not be modified
+# within individual charm repos. https://launchpad.net/bootstack-charms-spec
+
+name: PR workflow running lint checkers, unit and functional tests
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    branches: [ master, main ]
+    paths-ignore:
+      - '**.md'
+      - '**.rst'
+
+jobs:
+  pr:
+    uses: canonical/bootstack-actions/.github/workflows/pull-request.yaml@main
+    secrets: inherit
+    with:
+      python-version-unit: "['3.8', '3.10']"
+      python-version-func: "3.10"
+      tox-version: "<4"
+      juju-channel: "3.1/stable"
```

Comment:
Is it possible that we can test Juju 2.9 & 3.1 both? We need to support 2 versions at the same time for a while.

Reply:
Unfortunately the `pull-request.yaml` do not supports two juju-channels. What we can do is add this functionality, but it will bring lot of new complexity and since it is just for some time, I think it would be better simply have two "pr" jobs. e.g.
```yaml
jobs:
  pr:
    uses: canonical/bootstack-actions/.github/workflows/pull-request.yaml@main
    secrets: inherit
    with:
      python-version-unit: "['3.8', '3.10']"
      python-version-func: "3.10"
      tox-version: "<4"
      juju-channel: "3.1/stable"
      commands: "['FUNC_ARGS=\"--series bionic\" make functional', 'FUNC_ARGS=\"--series focal\" make functional', 'FUNC_ARGS=\"--series jammy\" make functional']"
3 changes: 1 addition & 2 deletions 
  pr-2.9:
    uses: canonical/bootstack-actions/.github/workflows/pull-request.yaml@main
    secrets: inherit
    with:
      python-version-unit: "[]"
      python-version-func: "3.10"
      tox-version: "<4"
      juju-channel: "2.9/stable"
      commands: "['FUNC_ARGS=\"--series bionic\" make functional', 'FUNC_ARGS=\"--series focal\" make functional', 'FUNC_ARGS=\"--series jammy\" make functional']"
3 changes: 1 addition & 2 deletions 
```

What do you say?

Reply:
As I know, charm-apt-mirror itself doesn't depends on libjuju. So it may be unnessary to run functional test both in juju 2.9 & 3. Let's keep it only in 3 now. If we find out new issues only for Juju 2.9 in the future, then we can make a change.